x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+//! A basic ERC-721 compatible contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use metadata::Nft;
+use wasmlanche_sdk::{program::Program, public, state_keys, types::Address};
+
+pub mod example;
+pub mod metadata;
+
+const NAME: &str = "My NFT";
+const SYMBOL: &str = "MNFT";
+const TOTAL_SUPPLY: u64 = 20;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
```

Comment:
If we're hardcoding this anyway, maybe we should remove it from the state-keys for now since there's no way to change it, right?

Reply:
imo it should be stored on-chain. This is the type of metadata that is helpful for users browsing NFTs.